cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java

```diff
@@ -168,25 +168,44 @@ private void setResultSetMetadata() throws Exception {
    *
    * @throws Exception if the driver is not available
    */
-  private void ensureJDBCDriverIsAvailable() throws Exception {
+  private void ensureJDBCDriverIsAvailable(Class<? extends Driver> jdbcDriverClass) throws Exception {
     try {
       DriverManager.getDriver(dbSinkConfig.connectionString);
     } catch (SQLException e) {
       // Driver not found. We will try to register it with the DriverManager.
       LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
-                dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, driverClass.getName(),
+                dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, jdbcDriverClass.getName(),
                 JDBCDriverShim.class.getName());
-      driverShim = new JDBCDriverShim(driverClass.newInstance());
-      DBUtils.deregisterAllDrivers(driverClass);
+      if (driverShim == null) {
+        driverShim = new JDBCDriverShim(jdbcDriverClass.newInstance());
+      }
+      DBUtils.deregisterAllDrivers(jdbcDriverClass);
```

Comment:
Unrelated to this PR - can you add the following comment to the above statement?
When JDBC Driver class is loaded, the driver class automatically registers itself with DriverManager. However, this driver is not usable due to different classloader used in plugins. Hence de-register this driver class, and any other driver classes registered by this classloader.